### PR TITLE
CONSOLE-3874: Refactor EventStream

### PR DESCRIPTION
**Pull request by TheRealJon**
Address tech debt in this component to make it simpler and less fragile.

### CODE REVIEW

File: frontend/public/components/events.jsx [hunk 1]
```diff
@@ -523,7 +513,7 @@ EventStream.propTypes = {
   textFilter: PropTypes.string,
 };
 
-EventStream.filterEvents = (messages, { kind, type, filter, textFilter }) => {
+const filterEvents = (messages, { kind, type, filter, textFilter }) => {
```
Comment (rhamilto) [inline frontend/public/components/events.jsx:516]: Nit:  move this to the top of the file with the other helpers?

**Review by rhamilto — APPROVED** [review]
/lgtm

Awesome work, @TheRealJon.

**Comment by yanpzhan** [comment]
Done a round of regression test on events related page. The test passed.

**Comment by rhamilto** [comment]
This is tech-debt and has no user-facing impact.

/label px-approved
/label docs-approved

**Pull request merged**